src/modules/Elsa.Workflows.Core/Contracts/IActivityRegistryUpdateService.cs

```diff
@@ -0,0 +1,32 @@
+namespace Elsa.Workflows.Contracts;
+
+/// <summary>
+/// Represents a service for updating the activity registry.
+/// </summary>
+public interface IActivityRegistryUpdateService
+{
+    /// <summary>
+    /// Tries to add a workflow as an activity to the registry.
+    /// </summary>
+    /// <param name="providerType">The type of the activity provider.</param>
+    /// <param name="workflowDefinitionId">The ID of the workflow definition.</param>
+    /// <param name="cancellationToken">The cancellation token.</param>
+    Task AddToRegistry(Type providerType, string workflowDefinitionId, CancellationToken cancellationToken = default);
```

Comment:
Given that this service is specifically designed to update the Activity Registry with Activity Descriptors provided by the WorkflowDefinitionActivityProvider, we can remove the providerType parameter and instead hardcode its use (`typeof(WorkflowDefinitionActivityProvider)`) in the implementing class.

On that note, we should probably also rename this service (IActivityRegistryUpdateService) to something like `WorkflowDefinitionActivityRegistryUpdater`, given that it's not a general-purpose activity registry updater, but rather, a service that handles workflow definition changes by surgically updating the activity registry.